daemon/api_quotas.go

```diff
@@ -169,27 +174,37 @@ func postQuotaGroup(c *Command, r *http.Request, _ *auth.UserState) Response {
 		}
 		if err == servicestate.ErrQuotaNotFound {
 			// then we need to create the quota
-			if err := servicestateCreateQuota(st, data.GroupName, data.Parent, data.Snaps, quantity.Size(data.MaxMemory)); err != nil {
+			ts, err = servicestateCreateQuota(st, data.GroupName, data.Parent, data.Snaps, quantity.Size(data.MaxMemory))
+			if err != nil {
 				// XXX: dedicated error type?
 				return BadRequest(err.Error())
 			}
+			chgSummary = "Create quota group"
 		} else if err == nil {
 			// the quota group already exists, update it
 			updateOpts := servicestate.QuotaGroupUpdate{
 				AddSnaps:       data.Snaps,
 				NewMemoryLimit: quantity.Size(data.MaxMemory),
 			}
-			if err := servicestateUpdateQuota(st, data.GroupName, updateOpts); err != nil {
+			ts, err = servicestateUpdateQuota(st, data.GroupName, updateOpts)
+			if err != nil {
 				return BadRequest(err.Error())
 			}
+			chgSummary = "Update quota group"
 		}
 
 	case "remove":
-		if err := servicestateRemoveQuota(st, data.GroupName); err != nil {
+		var err error
+		ts, err = servicestateRemoveQuota(st, data.GroupName)
+		if err != nil {
 			return BadRequest(err.Error())
 		}
+		chgSummary = "Remove quota group"
 	default:
 		return BadRequest("unknown quota action %q", data.Action)
 	}
-	return SyncResponse(nil)
+
+	chg := newChange(st, "quota-control", chgSummary, []*state.TaskSet{ts}, data.Snaps)
+	st.EnsureBefore(0)
```

Comment:
we tend to use a mockable helper ensureStateSoon

Reply:
done